### ISSUE TITLE
Build error in Windows

### ISSUE THREAD

**Issue by Dipinka** [google/differential-privacy#18]
Receiving following error after running "bazel build differential_privacy/…"

<path>/differential-privacy/differential_privacy/base/testing/BUILD:38:1: C++ compilation of rule '//differential_privacy/base/testing:status_matchers' failed (Exit 2)
cl : Command line error D8021 : invalid numeric argument '/Wno-sign-compare'
INFO: Elapsed time: 137.683s, Critical Path: 20.36s
INFO: 157 processes: 157 local.
FAILED: Build did NOT complete successfully

**Comment by dasmdasm** [comment]
This is because we add a
`    copts = ["-Wno-sign-compare"],
`
line to every build rule. I guess that compiler flag isn't available on windows. It's not necessary, it just suppresses compiler warnings, so I'll look into making it platform-dependent. In the mean time, if you need a workaround you can remove it from the rules you're trying to build.

**Comment by aitsam12** [comment]
Is it available for windows now? I am still getting this error. 
If not, then how to remove it from rules.

**Comment by dasmdasm** [comment]
Nope, we have not made any progress on this yet. The workaround is to remove the `copts = ["-Wno-sign-compare"],` from the build rules you want to use on Windows.

**Comment by TedTed** [comment]
Do @BenjaminDev's patches solve this error, @Dipinka, or is the workaround still necessary?

**Comment by BenjaminDev** [comment]
@TedTed as far as I'm aware my patches solved the issue. PyDP have the windows build as part of the ci and no one is complaining there. Just a note that PyDP is normally a few commits behind.

@Dipinka if you getting build issues take a look at the[ GitHub Actions for PyDP](https://github.com/OpenMined/PyDP/blob/dev/.github/workflows/pypipublish_windows.yml) as it might shed some light on how we build it.

**Comment by jasmiem** [comment]
[GitHub Actions for PyDP](https://github.com/OpenMined/PyDP/blob/dev/.github/workflows/pypipublish_windows.yml), which is not exist

**Comment by BenjaminDev** [comment]
I've not been working on PyDP in a while and a lot has changed. If `google/differential-privacy` wants to support windows "officially" I'd suggest we add some form of CI that builds for windows - otherwise there will forever be breaks that go unnoticed until PyDP tries to do a release.

I'd be happy to have a go at making a `google/differential-privacy/.github/workflows/windows.yml` which does a windows build. Then we can catch breaks when they happen. Just want to be sure we want to support windows in this way i.e if it breaks windows we fix before landing on `main`.

**Comment by dibakch** [comment]
We have an internal CI job that runs on Windows.  However, it builds the C++ library without the `postgres` and `accounting` folders, i.e. using the following line:
> bazelisk build -- ... -postgres/... -accounting/...

Could you provide more information about the error?  Which commit of the library are you using and which targets cause the issue?

**Comment by BenjaminDev** [comment]
> We have an internal CI job that runs on Windows
Oh, cool. I just happened to do the initial work to get the windows builds going and thought I could assist. If CI is happy then all good. Thanks @dibakch
